packages/css/src/rules/types.ts

```diff
@@ -59,3 +59,133 @@ export type RulesVariants<RuleFn extends RuntimeFn<VariantGroups>> = Resolve<
 >;
 export type RecipeVariants<RecipeFn extends RuntimeFn<VariantGroups>> =
   RulesVariants<RecipeFn>;
+
+// == Tests ====================================================================
+if (import.meta.vitest) {
+  const { describe, it, assertType } = import.meta.vitest;
+
+  describe.concurrent("VariantSelection Type Test", () => {
+    type ExampleVariants = {
+      size: {
+        small: string;
+        large: string;
+      };
+      color: {
+        red: string;
+        blue: string;
+      };
+    };
+    function assertSelectedVariant(
+      variants: VariantSelection<ExampleVariants>
+    ) {
+      assertType<VariantSelection<ExampleVariants>>(variants);
+      return variants;
+    }
+
+    it("Valid VariantSelection Type", () => {
+      assertSelectedVariant({
+        size: "small",
+        color: "red"
+      });
+    });
+
+    it("Invalid VariantSelection Type", () => {
+      assertSelectedVariant({
+        // @ts-expect-error: selected variant value is not included in `size` key.
+        size: "medium", // error occured here
+        color: "red"
+      });
+    });
+  });
+
+  describe.concurrent("Compound Variants Type Test", () => {
+    function asseertCompoundVariants<Variants extends VariantGroups>(
+      optionValue: CompoundVariant<Variants>
+    ) {
+      assertType(optionValue);
+      return optionValue;
+    }
+
+    it("Vaild CompoundVariant", () => {
+      asseertCompoundVariants({
+        variants: {
+          color: "brand",
+          size: "small"
+        },
+        style: {
+          fontSize: "16px"
+        }
+      });
+    });
+
+    it("Invaild CompoundVariant with default key", () => {
+      asseertCompoundVariants({
+        // @ts-expect-error: selected variant key is not invalid
+        variantss: {
+          // ~~~~
+          // error occured here
```

Comment:
Do you  mean like this?

```typescript
        variantss: { // error occured here
```

Reply:
Yes right, I marked it temporarily because of the lint😂
But I'm planning to change it.